src/backchannel_logout.rs

```diff
@@ -0,0 +1,469 @@
+//! This module implements components needed for [back-channel logout]
+//!
+//! [back-channel logout]: <https://openid.net/specs/openid-connect-backchannel-1_0.html>
+
+use std::collections::HashMap;
+
+use chrono::{DateTime, Utc};
+use serde::{de::Error, Deserialize};
+
+use crate::{
+    types::helpers::{deserialize_string_or_vec, serde_utc_seconds},
+    types::SessionIdentifier,
+    Audience, IssuerUrl, SubjectIdentifier,
+};
+
+/// The Logout Token as defined in [section 2.4] of the [OpenID Connect Back-Channel Logout spec][1]
+///
+/// [section 2.4]: <https://openid.net/specs/openid-connect-backchannel-1_0.html#LogoutToken>
+/// [1]: <https://openid.net/specs/openid-connect-backchannel-1_0.html>
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct LogoutToken {
+    /// The issuer of this token
+    iss: IssuerUrl,
+    /// The audience this token is intended for
+    aud: Vec<Audience>,
+    /// Time at which this token was issued
+    iat: DateTime<Utc>,
+    /// The unique identifier for this token. This can be used to detect
+    /// replay attacks.
+    jti: String,
```

Comment:
For consistency with the rest of the crate, let's define a `JsonWebTokenId` newtype instead of using a plain string. That'll also help if this crate ever needs to support JWT-based client auth.